core/modules/user/user.module

```diff
@@ -2771,19 +2898,44 @@ function _user_mail_notify($op, $account, $language = NULL) {
 
 /**
  * Form element process handler for client-side password hide/show and strength.
- * @param $element
+ *
+ * @param array $element
+ *   The form element to process.
+ * @param array $form_state
+ *   The $form_state array of the form this element belongs to.
  */
-function user_form_process_password($element) {
+function user_form_process_password($element, $form_state = array()) {
   global $user;
+  $config = config('system.core');
 
   if ($element['#password_strength']) {
+    if (isset($form_state['user']) && is_a($form_state['user'], 'User')) {
+      // Existing account, not the current user if an admin edits an account.
+      $username = $form_state['user']->name;
+      $email = $form_state['user']->mail;
+    }
+    else {
+      // The AnonymousUser object has no mail property, and name is NULL.
+      // The User object has both mail and name set.
+      $username = (isset($user->name) ? $user->name : '');
+      $email = (isset($user->mail) ? $user->mail : '');
+    }
     $strength_settings = array(
-      'strengthTitle' => t('Password strength: '),
-      'weak' => t('poor'),
-      'fair' => t('fair'),
-      'good' => t('good'),
-      'strong' => t('excellent'),
-      'username' => (isset($user->name) ? $user->name : ''),
+      'labels' => array(
+        'strengthTitle' => t('Password strength: '),
```

Comment:
Symbols that are universal and especially trailing spaces should be avoided in translatable strings. This can instead be:

```suggestion
        'strengthTitle' => t('Password strength') . ': ',
```

or:

```suggestion
        'strengthTitle' => t('Password strength:') . ' ',
```

or just:

```suggestion
        'strengthTitle' => t('Password strength:'),
```

...and the `: ` bit can be appended to the `passwordStrengthSettings.labels.strengthTitle` in the JS.

Reply:
gotcha